plugin/latexlivepreview.vim

```diff
@@ -61,69 +62,121 @@ function! s:Compile()
         return
     endif
 
-    lcd %:p:h
+    " Change directory to handle properly sourced files with \input and bib
+    " TODO: get rid of lcd
+    execute 'lcd ' . b:livepreview_buf_data['root_dir']
 
+    " Write the current buffer in a temporary file
     silent exec 'write! ' . b:livepreview_buf_data['tmp_src_file']
 
-    call s:RunInBackground(
-                \ 'pdflatex -shell-escape -interaction=nonstopmode -output-directory=' .
-                \ b:livepreview_buf_data['tmp_dir'] . ' ' .
-                \ b:livepreview_buf_data['tmp_src_file'])
-
-    if b:livepreview_buf_data['has_bibliography']
-        " ToDo: Make the following work in Windows
-        call system('cd ' .  b:livepreview_buf_data['tmp_dir'] .
-                    \ ' && bibtex *.aux')
-    endif
+    call s:RunInBackground(b:livepreview_buf_data['run_cmd'])
 
     lcd -
 endfunction
 
-function! s:StartPreview()
-    lcd %:p:h
-    
+function! s:StartPreview(...)
     let b:livepreview_buf_data = {}
 
     " Create a temp directory for current buffer
     python << EEOOFF
 vim.command("let b:livepreview_buf_data['tmp_dir'] = '" +
-        tempfile.mkdtemp() + "'")
+        tempfile.mkdtemp(prefix="vim-latex-live-preview-") + "'")
 EEOOFF
 
     let b:livepreview_buf_data['tmp_src_file'] =
-                \ b:livepreview_buf_data['tmp_dir'] . '/' .
-                \ fnameescape(expand('%:r')) . '.' . expand('%:e')
+                \ b:livepreview_buf_data['tmp_dir'] .
+                \ fnameescape(expand('%:p:r') . '.' . expand('%:e'))
+
+    " Guess the root file which will be compiled, using first the argument
+    " passed, then the first line declaration of the source file and
+    " eventually fallback to the current file.
+    " TODO: emulate -parse-first-line properly
+    let l:root_line = substitute(getline(1),
+                \ '\v^\s*\%\s*!tex\s*root\s*\=\s*(.*)\s*$',
+                \ '\1', '')
+    if (a:0 > 0)
+        let l:root_file = fnameescape(fnamemodify(a:1, ':p'))
```

Comment:
I'm wondering what happens when `a:1` is a relative path: what is the parent directory then?
Answering this question will help to get rid of all the `lcd`.